### PR TITLE
script to convert ASCII data to netcdf files

### PR DESCRIPTION
**Pull request by alarshi**
this PR adds a python script that can use an input ASCII file structured using ASPECT convention and then convert it to netcdf format, which is much smaller in size. 
We also account for differences in the output netcdf for visualization in paraview and the aspect input file, in case of spherical coordinate system.

### Before your first pull request:

* [x] I have read the guidelines in our [CONTRIBUTING.md](../blob/main/CONTRIBUTING.md) document.

### For all pull requests:

* [ ] I have followed the [instructions for indenting my code](../blob/main/CONTRIBUTING.md#making-aspect-better).

### CODE REVIEW

File: contrib/python/scripts/ascii2netcdf.py [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+    current_argument = 2
+    file_name        = sys.argv[1]
+
+    # default argument values
+    dim               = 2
+    coordinate_system = 'cartesian'
+    output_format     = 'aspect'
+    
+    if (argc < 2):
+        raise Exception("Not enough arguments. Please provide the name of the input ascii file for conversion.")
```
Comment (gassmoeller) [inline contrib/python/scripts/ascii2netcdf.py:42]: Wouldnt you need to check this before line 34 where you access argv?

File: contrib/python/scripts/ascii2netcdf.py [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+    dim               = 2
+    coordinate_system = 'cartesian'
+    output_format     = 'aspect'
+    
+    if (argc < 2):
+        raise Exception("Not enough arguments. Please provide the name of the input ascii file for conversion.") 
+    
+    while (current_argument < argc):
+        if (args[current_argument] == '3' or args[current_argument] == '2'):
+            dim    = sys.argv[2]
```
Comment (gassmoeller) [inline contrib/python/scripts/ascii2netcdf.py:46]:
```suggestion
            dim = sys.argv[2]
```

File: contrib/python/scripts/ascii2netcdf.py [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+
+    coords_r     = np.unique(ascii_data[:, 0])
+    coords_phi   = np.unique(ascii_data[:, 1])
+    n_r          = len(coords_r)
+    n_phi        = len(coords_phi)
+    
+    coords = [ds.createDimension('radius', n_r),
+              ds.createDimension('longitude', n_phi)]
+    
+    # It is important to have follow the convention here in Paraview for
```
Comment (gassmoeller) [inline contrib/python/scripts/ascii2netcdf.py:122]:
```suggestion
    # It is important to follow the convention here in Paraview for 
```

**Review by gassmoeller — COMMENTED** [review]
Very nice, just a few small comments. Do you want to add a changelog entry too to make people aware of the script?

**Comment by alarshi** [comment]
Thank you ! I addressed your suggestions in the latest commit and added a changelog entry.

File: contrib/python/scripts/ascii2netcdf.py [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+import netCDF4 as nc
+import numpy as np
+import sys
+
+def main():
+
+    msg = "Usage            : ascii2netcdf.py input_ascii_filename  [dimensions, coordinate_system, output_format] \n \n" + \
+          "Optional arguments: \n" + \
+          "dimensions       : '2' or '3', the dimensions of the data in the ascii file \n" +  \
+          "coordinate_system: 'spherical' or 'cartesian' \n" +  \
```
Comment (tjhei) [inline contrib/python/scripts/ascii2netcdf.py:20]: What if the ASCII file is something different like a P-T table? Would a third option that just preserves the columns as is possible?

**Review by tjhei — APPROVED** [review]
Very nice!

File: contrib/python/scripts/ascii2netcdf.py (hunk 4 above)
Comment (alarshi) [reply contrib/python/scripts/ascii2netcdf.py:21]: Something like in the latest commit ? I added another function to take care of such files. I also forgot about the ascii depth profile files.

**Comment by tjhei** [comment]
I am not sure how specifying dim=0 would work. When loading, depth profiles are dim=1, p-t tables are dim=2, etc.. The data columns clearly need to depend on the other variables.
Is there a way for Python to grab the table headers from the ASCII file? If not, we can just ask the user to specify it if necessary.

Am I wrong?

**Comment by tjhei** [comment]
We can of course merge as is and improve later...

**Review by gassmoeller — APPROVED** [review]
Given that @alarshi is travelling for the next weeks, I would say let's merge this as is. We can improve it for special cases later, but it already works for one of the most common cases.

**Pull request merged**